go/vt/vtgate/evalengine/func.go

```diff
@@ -594,3 +595,29 @@ func aggregatedType(env *ExpressionEnv, expr []Expr) sqltypes.Type {
 	}
 	return sqltypes.VarChar
 }
+
+// builtinSign function
+type builtinSign struct{}
+
+func (builtinSign) call(env *ExpressionEnv, args []EvalResult, result *EvalResult) {
+	toSign := &args[0]
+	if toSign.null() {
+		result.setNull()
+	}
+
+	if toSign.float64() > 0 {
+		result.setInt64(1)
+	} else if toSign.float64() == 0 {
+		result.setInt64(0)
+	} else {
+		result.setInt64(-1)
+	}
+
+}
+
+func (builtinSign) typeof(env *ExpressionEnv, args []Expr) (sqltypes.Type, flag) {
+	if len(args) != 1 {
+		throwArgError("SIGN")
+	}
+	return args[0].typeof(env)
```

Comment:
This can't possibly be right, as the return of the `SIGN` function is not related to the actual input argument.